genai-perf/tests/test_json_exporter.py

```diff
@@ -682,3 +682,399 @@ def test_invalid_goodput_json_output(
+                        "min": 2.1527738520000033,
+                        "max": 2.1554224260000026
+                    }
+                },
+                "gpu_utilization": {
+                    "unit": "%",
+                    "gpu0": {
+                        "avg": 8.72972972972973,
+                        "p25": 9.0,
+                        "p50": 9.0,
+                        "p75": 9.0,
+                        "p90": 10.0,
+                        "p95": 10.0,
+                        "p99": 10.0,
+                        "min": 0.0,
+                        "max": 10.0
+                    }
+                },
+                "total_gpu_memory": {
+                    "unit": "GB",
+                    "gpu0": {
+                        "avg": 51.52702464
+                    }
+                },
+                "gpu_memory_used": {
+                    "unit": "GB",
+                    "gpu0": {
+                        "avg": 26.052919296000006,
+                        "p25": 26.052919296000002,
+                        "p50": 26.052919296000002,
+                        "p75": 26.052919296000002,
+                        "p90": 26.052919296000002,
+                        "p95": 26.052919296000002,
+                        "p99": 26.052919296000002,
+                        "min": 26.052919296000002,
+                        "max": 26.052919296000002
+                    }
+                }
+            }
+        """
+
+        cli_cmd = [
+            "genai-perf",
+            "profile",
+            "-m",
+            "gpt2_vllm",
+            "--service-kind",
+            "triton",
+            "--streaming",
+            "--server-metrics-url",
+            "http://tritonmetrics:8002/metrics",
+        ]
+
+        monkeypatch.setattr("sys.argv", cli_cmd)
+        args, _ = parser.parse_args()
+        config = ExporterConfig()
+        config.stats = stats
+        config.telemetry_stats = telemetry_stats
+        config.args = args
+        config.extra_inputs = parser.get_extra_inputs_as_dict(args)
+        config.artifact_dir = args.artifact_dir
+        json_exporter = JsonExporter(config)
+        assert json_exporter._stats_and_args["telemetry_stats"] == json.loads(
+            expected_telemetry_json_output
+        )
+
+        json_exporter.export()
+        expected_filename = "profile_export_genai_perf.json"
+        written_data = [
+            data
+            for filename, data in mock_read_write
+            if os.path.basename(filename) == expected_filename
+        ]
+        if written_data == []:
+            raise Exception(
+                f"Expected file {expected_filename} not found in written data."
+            )
+        assert len(written_data) == 1
+        output_data_dict = json.loads(written_data[0])
+
+        assert output_data_dict["telemetry_stats"] == json.loads(
+            expected_telemetry_json_output
+        )
+
+    def test_triton_telemetry_output_failure(
```

Comment:
same here. the test name "failure" makes it sounds like we need to catch some kind of error. I think more appropriate name would be something like `test_triton_telemetry_no_output`.